pkg/webhooks/clusterqueue_webhook.go

```diff
@@ -106,25 +112,24 @@ func ValidateClusterQueue(cq *kueue.ClusterQueue) field.ErrorList {
 		validation.ValidateLabelSelector(cq.Spec.NamespaceSelector, validation.LabelSelectorValidationOptions{}, path.Child("namespaceSelector"))...)
 	allErrs = append(allErrs, validateCQAdmissionChecks(&cq.Spec, path)...)
 	if cq.Spec.Preemption != nil {
-		allErrs = append(allErrs, validatePreemption(cq.Spec.Preemption, path.Child("preemption"))...)
+		allErrs = append(allErrs, w.validatePreemption(cq.Spec.Preemption, path.Child("preemption"))...)
 	}
 	if cq.Spec.FairSharing != nil {
 		allErrs = append(allErrs, validateFairSharing(cq.Spec.FairSharing, path.Child("fairSharing"))...)
 	}
 	return allErrs
 }
 
-func ValidateClusterQueueUpdate(newObj *kueue.ClusterQueue) field.ErrorList {
-	return ValidateClusterQueue(newObj)
-}
-
-func validatePreemption(preemption *kueue.ClusterQueuePreemption, path *field.Path) field.ErrorList {
+func (w *ClusterQueueWebhook) validatePreemption(preemption *kueue.ClusterQueuePreemption, path *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
 	if preemption.ReclaimWithinCohort == kueue.PreemptionPolicyNever &&
 		preemption.BorrowWithinCohort != nil &&
 		preemption.BorrowWithinCohort.Policy != kueue.BorrowWithinCohortPolicyNever {
 		allErrs = append(allErrs, field.Invalid(path, preemption, "reclaimWithinCohort=Never and borrowWithinCohort.Policy!=Never"))
 	}
+	if w.fairSharingEnabled && preemption.BorrowWithinCohort != nil && preemption.BorrowWithinCohort.Policy != kueue.BorrowWithinCohortPolicyNever {
+		allErrs = append(allErrs, field.Invalid(path, preemption.BorrowWithinCohort, "FairSharing incompatible with borrowWithinCohort"))
```

Comment:
```suggestion
		allErrs = append(allErrs, field.Invalid(path.Child("borrowWithinCohort").Child("policy"), preemption.BorrowWithinCohort.Policy, "FairSharing incompatible with borrowWithinCohort except for Never policy"))
```

Actually, this incompatible came from policy, right?